### ISSUE TITLE
[Feature] Example of wpbp_get_template_part and get_template_part

### ISSUE THREAD

**Issue by MikeiLL** [WPBP/WordPress-Plugin-Boilerplate-Powered#165]
I want to include a CPT template file that calls on template partials.

Your `wpbp_get_template_part()` looks great, but I'm a little confused on how to use it.

Currently I load a singular cpt template with filters in the Template file:

```
add_filter( 'single_template', array( __CLASS__, 'load_single_od_video' ) );
public static function load_single_od_video( $single_template ) {
    global $post;

    if ( 'elliptica_od_video' === $post->post_type ) {
        return wpbp_get_template_part( MMC_TEXTDOMAIN, 'single', 'ondemand', false );
    }

    return $single_template;
}
```

In `templates/single-ondemand.php` I can pull in the partial:

1.
```
ob_start();
get_template_part( 'content', 'ondemand' );
echo ob_get_clean();
```

But I expect that is not the intended approach, either within a page template or within a shortcode.

**Comment by Mte90** [comment]
There is an example on how to use it https://github.com/WPBP/WordPress-Plugin-Boilerplate-Powered/blob/master/plugin-name/integrations/Template.php

My version is a different version that will look for more file and return that, so it's the same to use the WP one.
Usually in theme files `ob_*` are avoided for performance reasons and I don't think that you need that.

I invite you to open the ticket in the right repo the next time https://github.com/WPBP/template

Anyway I don't understand what issue do you have.

**Comment by MikeiLL** [comment]
I had thought about opening the ticket on the [wpbp/template](https://github.com/WPBP/template) repo, but what I'm requesting is an example of, here in the BP, of how I would load template_part in this system. I can call `wpbp_get_template_part()`, which returns the path to the template, but I'm not sure how to make it display on the page.

It would be cool if this repository had an example of that in the content-demo template, or perhaps we could also add an example of filtering single_template or page_template.

I understand if these are things that developers are expected to figure out on their own, however I think there are a lot of great practices included here and the easier it is for less educated developers to use it, the wider it's influence can be.

**Comment by Mte90** [comment]
Right now it is enough to use the example that is already in the boilerplate https://github.com/WPBP/WordPress-Plugin-Boilerplate-Powered/blob/master/plugin-name/integrations/Template.php?rgh-link-date=2020-08-03T11%3A26%3A00Z

Because this is to be used inside the plugins to load different files and let them to add it in the theme folder.

The `templates/single-ondemand.php` is inside the plugin not the theme so you need to customize as you need for that purpose. I guess that ``get_template_part( 'content', 'ondemand' );` is calling another file that is inside the theme instead.
So the right flow it is:

* create a folder inside the theme `plugin-name`
* now you can put all the files that you want to customize the plugin

Basically this library is to copy what is possible to do with woocommerce and their template files and the plugin magically manage that (I will improve a bit the readme of that library).

**Comment by MikeiLL** [comment]
Thanks. What I'm familiar with is [Gamajo Template Loader](https://github.com/GaryJones/Gamajo-Template-Loader), where you are able to have templates provided by Plugin, over-ridable in theme.

Sound like this is similar. I'll continue to experiment with it.

**Comment by Mte90** [comment]
I improved a bit the readme if you have any suggestions about how to improve the docs let me know.
